includes/abstracts/abstract-wc-payment-token.php

```diff
@@ -0,0 +1,271 @@
+	 * @param boolean $is_default True or false
+	 */
+	public function set_default( $is_default ) {
+		$this->data['is_default'] = (bool) $is_default;
+	}
+
+	/**
+	 * Returns a dump of the token data (combined data and meta)
+	 *
+	 * @since 2.6.0
+	 * @return mixed array representation
+	 */
+	public function get_data() {
+		return array_merge( $this->data, array( 'meta' => $this->meta ) );
+	}
+
+	/**
+	 * Validate basic token info (token and type are required)
+	 *
+	 * @since 2.6.0
+	 * @return boolean True if the passed data is valid
+	 */
+	public function validate() {
+		if ( empty( $this->data['token'] ) ) {
+			return false;
+		}
+
+		if ( empty( $this->data['type'] ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get a token from the database
+	 *
+	 * @since 2.6.0
+	 * @param  int $token_id Token ID
+	 */
+	public function read( $token_id ) {
+		global $wpdb;
+		if ( $token = $wpdb->get_row( $wpdb->prepare( "SELECT * FROM {$wpdb->prefix}woocommerce_payment_tokens WHERE token_id = %d LIMIT 1;", $token_id ) ) ) {
+			$this->id = $token->token_id;
+			$token = (array) $token;
+			unset( $token['token_id'] );
+			$this->data = $token;
+			$meta =  get_metadata( 'payment_token', $token_id );
+			$passed_meta = array();
+			if ( ! empty( $meta ) ) {
+				foreach( $meta as $meta_key => $meta_value ) {
+					$passed_meta[ $meta_key ] = $meta_value[0];
+				}
+			}
+			$this->meta = $passed_meta;
+		}
+	}
+
+	/**
+	 * Update a payment token
+	 *
+	 * @since 2.6.0
+	 * @return True on success, false if validation failed and a payment token could not be updated
+	 */
+	public function update() {
+		if ( false === $this->validate() ) {
+			return false;
+		}
+
+		global $wpdb;
+
+		$wpdb->update( $wpdb->prefix . 'woocommerce_payment_tokens', $this->data, array( 'token_id' => $this->get_id() ) );
+		foreach ( $this->meta as $meta_key => $meta_value ) {
+			update_metadata( 'payment_token', $this->get_id(), $meta_key, $meta_value );
+		}
+
+		do_action( 'woocommerce_payment_token_updated', $this->get_id() );
+		return true;
+	}
+
+	/**
+	 * Create a new payment token in the database
+	 *
+	 * @since 2.6.0
+	 * @return True on success, false if validation failed and a payment token could not be created
+	 */
+	public function create() {
+		if ( false === $this->validate() ) {
+			return false;
+		}
+
+		global $wpdb;
+
+		$wpdb->insert( $wpdb->prefix . 'woocommerce_payment_tokens', $this->data );
+		$this->id = $token_id = $wpdb->insert_id;
+		foreach ( $this->meta as $meta_key => $meta_value ) {
+			add_metadata( 'payment_token', $token_id, $meta_key, $meta_value, true );
+		}
+
+		do_action( 'woocommerce_payment_token_created', $token_id );
+		return true;
+	}
+
+	/**
+	 * Saves a payment token to the database - does not require you to know if this is a new token or an update token
+	 *
+	 * @since 2.6.0
+	 * @return True on success, false if validation failed and a payment token could not be saved
+	 */
+	public function save() {
```

Comment:
Save should return `$this->get_id()` to match what orders are doing in the order CRUD branch.